tests/integration/linux_bridge_test.py

```diff
@@ -139,6 +148,14 @@ def test_remove_bridge_and_keep_slave_up(bridge0_with_port0, port0_up):
     assert 1 == len(current_state[Interface.KEY])
 
 
+def test_create_vlan_as_slave_of_linux_bridge(setup_vlan_iface):
+    bridge_name = TEST_BRIDGE0
+    port_name = setup_vlan_iface[Interface.KEY][0][Interface.NAME]
+    bridge_state = _create_bridge_subtree_config((port_name,))
+    with _linux_bridge(bridge_name, bridge_state):
+        assertlib.assert_state(setup_vlan_iface)
```

Comment:
This verifies that the vlan is included in the current state.
But we are not interested to check about vlan creation here. We are interested to check if we can attach a vlan to the bridge. So it makes sense to check against the bridge desired state.